test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1

```diff
@@ -75,35 +77,46 @@ Describe "Clear-Content cmdlet tests" -Tags "CI" {
       $cci.SupportsShouldProcess | Should -BeTrue
     }
 
-    It "Alternate streams should be cleared with clear-content" -Skip:(!$IsWindows) {
-      # make sure that the content is correct
-      # this is here rather than BeforeAll because only windows can write to an alternate stream
-      Set-Content -Path "TestDrive:/$file3" -Stream $streamName -Value $streamContent
-      Get-Content -Path "TestDrive:/$file3" | Should -BeExactly $content2
-      Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeExactly $streamContent
-      Clear-Content -Path "TestDrive:/$file3" -Stream $streamName
-      Get-Content -Path "TestDrive:/$file3" | Should -BeExactly $content2
-      Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeNullOrEmpty
-    }
-
-    It "the '-Stream' dynamic parameter is visible to get-command in the filesystem" -Skip:(!$IsWindows) {
-      try {
-        Push-Location -Path TestDrive:
-        (Get-Command Clear-Content -Stream foo).parameters.keys -eq "stream" | Should -Be "stream"
+    Context "Clear-Content should work with alternate data streams on Windows" {
+      It "Alternate streams should be cleared with Clear-Content on a file" -Skip:(!$IsWindows) {
+        # Make sure the test is set up correctly.
+        Set-Content           -Path "TestDrive:/$file3" -Stream $streamName -Value $streamContent
+        Get-Content           -Path "TestDrive:/$file3" -Stream $streamName | Should -BeExactly $streamContent
+        # Truncate the stream.
+        Clear-Content         -Path "TestDrive:/$file3" -Stream $streamName -ErrorAction Stop
+        # The stream should exist, but should have 0 length.
+        $result = Get-Item -Path "TestDrive:/$file3" -Stream $streamName
+        $result | Should -Not -BeNullOrEmpty
+        $result.length | Should -Be 0
+      }
+      It "Alternate streams should be cleared with Clear-Content on a directory" -Skip:(!$IsWindows) {
+        # Make sure the test is set up correctly.
+        Set-Content           -Path "TestDrive:/$dirName" -Stream $streamName -Value $streamContent
+        # Truncate the stream.
+        Get-Content           -Path "TestDrive:/$dirName" -Stream $streamName | Should -BeExactly $streamContent
+        Clear-Content         -Path "TestDrive:/$dirName" -Stream $streamName -ErrorAction Stop
+        # The stream should exist, but should have 0 length.
+        $result = Get-Item -Path "TestDrive:/$dirName" -Stream $streamName
+        $result | Should -BeOfType System.Management.Automation.Internal.AlternateStreamData
+        $result.length | Should -Be 0
+      }
+      It "the '-Stream' dynamic parameter is visible to get-command in the filesystem" -Skip:(!$IsWindows) {
+        try {
+          Push-Location -Path TestDrive:
+          (Get-Command Clear-Content -Stream foo).parameters.keys -eq "stream" | Should -Be "stream"
```

Comment:
```suggestion
          (Get-Command Clear-Content -Stream foo).parameters.keys -eq "Stream" | Should -BeExactly "Stream"
```

Reply:
I agree with you that this should be fixed.  However, I thought you said that changes to things outside of what the PR directly relates to should be put into a different PR?  I never touched the detection of the parameters in my efforts to make sure that -Stream works with directories.

Reply:
Yes, but we changed the code so it is good to make the minor fix for consistency.